### PR TITLE
[DependencyInjection] Add `#[AutowireLocator]` attribute

### LINKED ISSUE #18774: [DependencyInjection] add `#[AutowireLocator]` attribute
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#51392
| PR author(s) | @kbond
| Merged in    | 6.4

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 6.4 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fix https://github.com/symfony/symfony-docs/issues/18774

### CODE REVIEW

File: service_container/service_subscribers_locators.rst [hunk 1]
```diff
@@ -297,6 +297,84 @@ This is done by having ``getSubscribedServices()`` return an array of
...
+    // src/CommandBus.php
+    namespace App;
+
+    use App\CommandHandler\BarHandler;
+    use App\CommandHandler\FooHandler;
+    use Psr\Container\ContainerInterface;
+    use Symfony\Component\DependencyInjection\Attribute\AutowireLocator;
+    use Symfony\Contracts\Service\ServiceSubscriberInterface;
+
+    class CommandBus implements ServiceSubscriberInterface
```
Comment (kbond) [inline service_container/service_subscribers_locators.rst]:
```suggestion
    class CommandBus
```

This should be removed, correct? (Same with below)

File: service_container/service_subscribers_locators.rst [hunk 2]
```diff
@@ -297,6 +297,84 @@ This is done by having ``getSubscribedServices()`` return an array of
...
+attribute::
+
+    // src/CommandBus.php
+    namespace App;
+
+    use App\CommandHandler\BarHandler;
+    use App\CommandHandler\FooHandler;
+    use Psr\Container\ContainerInterface;
+    use Symfony\Component\DependencyInjection\Attribute\AutowireLocator;
+    use Symfony\Contracts\Service\ServiceSubscriberInterface;
```
Comment (kbond) [inline service_container/service_subscribers_locators.rst]:
```suggestion
```

**Review by kbond — CHANGES_REQUESTED** [review]
Thanks for this @alexandre-daubois!

File: service_container/service_subscribers_locators.rst (hunk 1 above)
Comment (alexandre-daubois) [reply service_container/service_subscribers_locators.rst]: Indeed! That's updated :slightly_smiling_face:

**Review by kbond — APPROVED** [review]

File: service_container/service_subscribers_locators.rst [hunk 3]
```diff
@@ -297,6 +297,82 @@ This is done by having ``getSubscribedServices()`` return an array of
...
+            if ($this->locator->has($commandClass)) {
+                $handler = $this->locator->get($commandClass);
+
+                return $handler->handle($command);
+            }
+        }
+    }
+
+Just like with the ``getSubscribedServices()`` method, it is possible
+to define aliased services thanks to named argument, as well as optional
```
Comment (nicolas-grekas) [inline service_container/service_subscribers_locators.rst]:
```suggestion
to define aliased services thanks to named arguments, as well as optional
```

**Review by nicolas-grekas — COMMENTED** [review]
LGTM thanks

File: service_container/service_subscribers_locators.rst (hunk 3 above)
Comment (alexandre-daubois) [reply service_container/service_subscribers_locators.rst]: Updated, thanks :slightly_smiling_face:

**Comment by javiereguiluz** [comment]
Another nice new feature and another perfect contribution. Thanks Alex ... and thanks to reviewers too!

**Pull request merged**